openedx/features/badging/tests/test_views.py

```diff
@@ -55,11 +54,9 @@ def test_trophycase_with_some_earned_badges(self, mock_populate_trophycase):
         :param mock_populate_trophycase: mock to assert it has been called with expected input
         :return: None
         """
-        request = self.request_factory.get(reverse('trophycase'), {'json': True})
-        request.user = self.user
-
+        self.client.login(username=self.user.username, password='test')
```

Comment:
Instead of making client login in each test separately move this line into `setUp`. If in any test you want unauthenticated client then you can log it out first `self.client.logout()`

Reply:
Done